engine.go

```diff
@@ -1,16 +1,121 @@
 package fuego
 
-func NewEngine() *Engine {
-	return &Engine{
-		OpenAPI:      NewOpenAPI(),
-		ErrorHandler: ErrorHandler,
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log/slog"
+	"os"
+	"path/filepath"
+)
+
+func NewEngine(options ...func(*Engine)) *Engine {
+	e := &Engine{
+		OpenAPI:       NewOpenAPI(),
+		OpenAPIConfig: defaultOpenAPIConfig,
+		ErrorHandler:  ErrorHandler,
+	}
+	for _, option := range options {
+		option(e)
 	}
+	return e
 }
 
 // The Engine is the main struct of the framework.
 type Engine struct {
-	OpenAPI      *OpenAPI
-	ErrorHandler func(error) error
+	OpenAPI       *OpenAPI
+	ErrorHandler  func(error) error
+	OpenAPIConfig OpenAPIConfig
 
 	acceptedContentTypes []string
 }
+
+type OpenAPIConfig struct {
+	// If true, the server will not serve nor generate any OpenAPI resources
+	Disabled bool
```

Comment:
We will also disable generation at the route registration level (and keep the checks) for faster server start-time (even if it'll save only a few microseconds)

Reply:
see #312